src/blocklock/BlocklockSender.sol

```diff
@@ -70,48 +119,117 @@ contract BlocklockSender is
         decryptionSender = IDecryptionSender(_decryptionSender);
     }
 
-    // OVERRIDDEN UPGRADE FUNCTIONS
-    function _authorizeUpgrade(address newImplementation) internal override onlyOwner {}
-
-    /**
-     * @dev See {IBlocklockSender-requestBlocklock}.
-     */
-    function requestBlocklock(uint256 blockHeight, TypesLib.Ciphertext calldata ciphertext)
-        external
-        returns (uint256)
-    {
+    /// @dev Overridden upgrade authorization function to ensure only an authorized caller can authorize upgrades.
+    function _authorizeUpgrade(address newImplementation) internal override onlyAdmin {}
+
+    /// @notice Requests a blocklock for a specified block height with the provided ciphertext and subscription ID
+    /// @param callbackGasLimit The gas limit allocated for the callback execution after the blocklock request
+    /// @param subId The subscription ID associated with the request
+    /// @param blockHeight The block height at which the blocklock is requested
+    /// @param ciphertext The ciphertext that will be used in the blocklock request
+    /// @return requestID The unique identifier for the blocklock request
+    /// @dev This function allows users to request a blocklock for a specific block height. The blocklock is associated with a given subscription ID
+    ///      and requires a ciphertext to be provided. The function checks that the contract is configured and not disabled before processing the request.
+    function requestBlocklock(
+        uint32 callbackGasLimit,
+        uint256 subId,
```

Comment:
I'd suggest we have two separate funcs for this, even if the other internally calls `requestBlocklock(someLimit, 0, someHeight, someCiphertext)` because I can already imagine builders getting confused by subId

Reply:
Good idea. It's been added
thanks @CluEleSsUK